common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -99,15 +100,17 @@ public MLAgent(StreamInput input) throws IOException{
         if (input.readBoolean()) {
             memory = new MLMemorySpec(input);
         }
-        createdTime = input.readInstant();
-        lastUpdateTime = input.readInstant();
-        appType = input.readString();
+        createdTime = input.readOptionalInstant();
+        lastUpdateTime = input.readOptionalInstant();
+        appType = input.readOptionalString();
```

Comment:
isn't app type mandatory for memory? or this is something different?

Reply:
It's different, we have memory type for memory